### PR TITLE
Support for Duo Trusted Endpoints

### PR DESCRIPTION
**Pull request by 6f6d6172**
This is almost entirely @ctennis's code with a little cleanup. See #540 for more context.

Effectively, if Duo Trusted Endpoints is enabled, the client (usually the browser, but saml2aws in this case) talks to a daemon locally via HTTP. I've tested Duo Push/Passcode + Trusted Endpoints and it works great. I have not yet tested u2f support.

### CODE REVIEW

**Comment by 6f6d6172** [comment]
(only opened it so I could see codecov output, it's still wip)

**Review by mapkon — CHANGES_REQUESTED** [review]
This needs work:
- Tests are failing
- There are deprecated methods pulled through
- And the coverage will most likely go down

File: .goreleaser.ubuntu-latest.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ builds:
   goos:
     - windows
     - linux
+    - darwin
```
Comment (mapkon) [inline .goreleaser.ubuntu-latest.yml:15]: What is the reasoning for this?

File: Makefile [hunk 2]
```diff
@@ -11,10 +11,22 @@ SOURCE_FILES?=$$(go list ./... | grep -v /vendor/)
 TEST_PATTERN?=.
 TEST_OPTIONS?=
 
-BIN_DIR := $(CURDIR)/bin
+BIN_DIR := ./bin
```
Comment (mapkon) [inline Makefile:14]: I am not sure I see the difference here, other than hard coding the current directory as the reference for searching the bin directory. What is the purpose of this change?

**Comment by mapkon** [comment]
@6f6d6172 This is not quite there mate. Please take another look, thanks!

**Comment by 6f6d6172** [comment]
@mapkon  Yup! Just working on this between tasks at work :)

**Comment by 6f6d6172** [comment]
@mapkon  Yup! Just working on this between tasks at work :) I just got tests running locally.

**Comment by lizduty** [comment]
confirming this also works for me @6f6d6172 please let me know if you need help!

**Comment by lizduty** [comment]
@6f6d6172 thank you for your work on this! Couldn't have taken it across the finish line without you and @ctennis

**Review by mapkon — COMMENTED** [review]
Is this still being considered or should we close the PR?

**Pull request closed without merging**